### ISSUE TITLE
App packet heap corruption

### ISSUE THREAD

**Issue by wowaser** [ultravideo/uvgRTP#127]
Hello, 

I've been experiencing heap corruption while using APP packets. This seems to be related to the length of the app packet. Please see the code attached below to reproduce the error. The string is generated by openCV filestorage in JSON format.
The crash happens on the sending of the APP packet.

The code below causes heap corruption. If you remove the "ASDASDASDASD" part from the string, it does not cause any error. 

```cpp

#include <uvgrtp/lib.hh>
#include <iostream>

void receiverSideAppHook(std::shared_ptr<uvgrtp::frame::rtcp_app_packet> packet)
{
	//std::cout << "receiver received a PACKET with NAME: " << packet->name << std::endl;
	std::string json = (char*)packet->payload;
	std::cout << "le json: " << json << std::endl;
}

int main() {

	uvgrtp::context ctx;
	uvgrtp::session* sess = ctx.create_session("127.0.0.1");

	int flags = RCE_RTCP | RCE_FRAGMENT_GENERIC;

	uvgrtp::media_stream* send = sess->create_stream(8000, 8080, RTP_FORMAT_GENERIC, flags);
	uvgrtp::media_stream* recv = sess->create_stream(8080, 8000, RTP_FORMAT_GENERIC, flags);

	std::function<void(std::shared_ptr<uvgrtp::frame::rtcp_app_packet>)> rfp;
	rfp = std::bind(&receiverSideAppHook, std::placeholders::_1);
	recv->get_rtcp()->install_app_hook(rfp);

	char* message = (char*)"Hello, world!";
	size_t msg_len = strlen(message) + 1;

	const char* appName = "JSN";
	std::string payload = "Hello, world!";
	uint8_t* pLoad = (uint8_t*)payload.c_str();
	int counter = 0;
	bool hasRunOnce = false;

	while (true) {
		send->push_frame((uint8_t*)message, msg_len, 0);

		if (counter % 10 == 0) {
			
			std::string app = "{\n    \"TYPE\": \"NV_ENC\",\n    \"BITRATE\": 1.0,\n    \"GOP\": 5,\n    \"FPS\": 24,\n    
                         \"AUTOFILL_FLAG\": 0,\n    \"TUNING\": \"lowlatencyASDASDASDASDASDASDASDASD\",\n    \"PRESET\": \"p5\"\n}\n";

			uint8_t* load = (uint8_t*)app.c_str();
			size_t len = app.length();

			send->get_rtcp()->send_app_packet(appName, 1, len, load);
		}
		
		std::this_thread::sleep_for(std::chrono::milliseconds(33));
		auto frm = recv->pull_frame(10);

		std::cout << (char*)frm->payload << std::endl;
		uvgrtp::frame::dealloc_frame(frm);
		++counter;
	}

}
```

Thanks

**Comment by wowaser** [comment]
This example doesn't have a "weird" string in it
```cpp
void receiverSideAppHook(std::shared_ptr<uvgrtp::frame::rtcp_app_packet> packet)
{
	//std::cout << "receiver received a PACKET with NAME: " << packet->name << std::endl;
	std::string json = (char*)packet->payload;
	std::cout << "le json: " << json << std::endl;
}

int main() {

	uvgrtp::context ctx;
	uvgrtp::session* sess = ctx.create_session("127.0.0.1");

	int flags = RCE_RTCP | RCE_FRAGMENT_GENERIC;

	uvgrtp::media_stream* send = sess->create_stream(8000, 8080, RTP_FORMAT_GENERIC, flags);
	uvgrtp::media_stream* recv = sess->create_stream(8080, 8000, RTP_FORMAT_GENERIC, flags);

	std::function<void(std::shared_ptr<uvgrtp::frame::rtcp_app_packet>)> rfp;
	rfp = std::bind(&receiverSideAppHook, std::placeholders::_1);
	recv->get_rtcp()->install_app_hook(rfp);

	char* message = (char*)"Hello, world!";
	size_t msg_len = strlen(message) + 1;

	const char* appName = "JSN";
	std::string payload = "Hello, world!";
	uint8_t* pLoad = (uint8_t*)payload.c_str();
	int counter = 0;
	bool hasRunOnce = false;
	std::string app = "s";
	while (true) {
		send->push_frame((uint8_t*)message, msg_len, 0);

		if (counter % 10 == 0) {
			
			app += "a";
			uint8_t* load = (uint8_t*)app.c_str();
			size_t len = app.length();
			std::cout << len << std::endl;
			send->get_rtcp()->send_app_packet(appName, 1, len, load);
		}
		
		std::this_thread::sleep_for(std::chrono::milliseconds(33));
		auto frm = recv->pull_frame(10);

		//std::cout << (char*)frm->payload << std::endl;
		uvgrtp::frame::dealloc_frame(frm);
		++counter;
	}

}
```

**Comment by wowaser** [comment]
Fixed by #128
